File: LogiCode-Client/src/Class/Student.ts
import { StudentCours, StudentLesson } from "./StudentLesson";
import { Task } from "./Task";

export class Student {

    constructor(
        public id?: number,
        public email?: string,
        public password?: string,
        public name?: string,
        public provider?: string,
        public providerId?: string,
        public potoUrl?: string,
        public studentCourses?: Array<StudentCours| null>,
        public studentLessons?: Array<StudentLesson | null>,
        public studentTasks?: Array<Task | null>){ 
    }
}

File: LogiCode-Server/core/DAO/Student.cs
using Core.DAO;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Core.DAO
{
    public class Student 
    {
        public int Id { get; set; }
        public string? GoogleId { get; set; }
        public string Name { get; set; }
        public string Email { get; set; }
        public string PotoUrl { get; set; }
        public ICollection<StudentCourse> StudentCourses { get; set; }
        public ICollection<StudentLesson> StudentLessons { get; set; }
        public ICollection<Core.DAO.MyTask> StudentTasks { get; set; }


    }
}
                                                                            